src/main/kotlin/gropius/model/user/IMSUser.kt

```diff
@@ -0,0 +1,46 @@
+package gropius.model.user
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLIgnore
+import gropius.model.architecture.IMS
+import io.github.graphglue.model.Direction
+import io.github.graphglue.model.DomainNode
+import io.github.graphglue.model.FilterProperty
+import io.github.graphglue.model.NodeRelationship
+import org.springframework.data.annotation.Transient
+
+@DomainNode
+@GraphQLDescription(
+    """A user an IMS.
+    This user might be linked to a GropiusUser.
+    Note that this link can change at any time.
+    The username might not be unique.
+    It is possible that this user never heard of Gropius, and is only known to the system due to sync adapters.
+    """
+)
+class IMSUser(
+    displayName: String,
+    email: String?,
+    @GraphQLIgnore
+    var username: String?
+) : User(displayName, email) {
+
+    companion object {
+        const val GROPIUS_USER = "GROPIUS_USER"
+    }
+
+    @GraphQLDescription("The username of the IMSUser. Synced from the IMS this user is part of. Might not be unique.")
+    override fun username(): String? = username
+
+    @NodeRelationship(GROPIUS_USER, Direction.OUTGOING)
+    @GraphQLDescription("The GropiusUser this IMSUser is linked to. An IMSUser might be linked to no GropiusUser.")
+    @FilterProperty
+    @delegate:Transient
+    var gropiusUser by NodeProperty<GropiusUser?>()
+
+    @NodeRelationship(IMS.USER, Direction.INCOMING)
+    @GraphQLDescription("The IMS this user is part of.")
+    @FilterProperty
+    @delegate:Transient
+    var ims by NodeProperty<IMS>()
```

Comment:
Should we allow one `IMSUser` to belong to multiple`IMS`? e.g. if we have a company gitea and gitlab and we automatically link the users there already?

Reply:
As those are closely linked to the account management in our authentication service, I'm not sure if this is a good idea as it might lead to a lot of additional complexity. As the author of the authentication service, has @ModellbahnFreak an opinion on this?

Reply:
How and when do you suggest using one IMS user for multiple IMS?

Do you mean if the company uses a single sign on for theier systems?

Even in the domain model (independant of login service) I think it's worth using an IMSUser per IMS as that allows the frontend to for example offer a link to that user/that users comment etc in THAT specific IMS that he created the comment etc. in

Reply:
I would agree on this.
Especially as the IMSUser also shows e.g. for comments where the comment was originally created.